pkg/client/client.go

```diff
@@ -205,3 +215,44 @@ var addProviderVersionToUserAgent = request.NamedHandler{
 	Name: "openshift.io/cluster-api-provider-aws",
 	Fn:   request.MakeAddToUserAgentHandler("openshift.io cluster-api-provider-aws", version.Version.String()),
 }
+
+func resolveEndpoints(awsConfig *aws.Config, ctrlRuntimeClient client.Client, region string) error {
+	infra := &configv1.Infrastructure{}
+	infraName := client.ObjectKey{Name: globalInfrastuctureName}
+
+	if err := ctrlRuntimeClient.Get(context.Background(), infraName, infra); err != nil {
+		return err
+	}
+
+	// Do nothing when custom endpoints are missing
+	if infra.Status.PlatformStatus == nil && infra.Status.PlatformStatus.AWS == nil {
+		return nil
+	}
+
+	customEndpointsMap := buildCustomEndpointsMap(infra.Status.PlatformStatus.AWS.ServiceEndpoints)
+
```

Comment:
nit: you can probably just return nil here when the len of map is zero as there is not custom resolution required.